app/Filters/AppointmentFilters.php

```diff
@@ -0,0 +1,15 @@
+<?php
+namespace App\Filters;
+
+use App\Enums\AppointmentStatus;
+
+class AppointmentFilters extends QueryFilters
+{   
+    public function time($time)
+    {
+        $from = $time;
+        $to = $time + (request()->duration * 60000);
+        $checkBefore = 120 * 60000;
```

Comment:
It's not a good idea to check if doctor is not having any appointments before the incoming appointment as he can have two or more following appointment without any problem.

Just check dates overlapping between the incoming appointment and the doctor current appointments:
https://stackoverflow.com/questions/325933/determine-whether-two-date-ranges-overlap

Reply:
done